stdlib/Printf/src/Printf.jl

```diff
@@ -762,6 +849,48 @@ const UNROLL_UPTO = 16
     return pos
 end
 
+
+@inline function findonesplace(buf, lbound, rbound)
+    decimalpoint = findlast(==(UInt8('.')), buf[lbound:rbound])
+    # find last digit of rounded float
+    if isnothing(decimalpoint)
+        for i in lbound:(rbound - 1)
+            if 0x30 <= buf[i] <= 0x39 && buf[i + 1] == UInt(' ')
+                decimalpoint = i + 1
+                break
+            end
+        end
+    end
+    onesplace = isnothing(decimalpoint) ? rbound : decimalpoint - 1
+    return onesplace
+end
+
+
+function insertsep(buf, headpos, numsep, onesplace; numskip=0)
+    intlength = (onesplace - headpos + 1) - numsep - numskip
+
+    headdivlength = mod1(intlength, 3) + numskip
+    seconddiv = headpos + numsep + headdivlength
+    separation = headpos + headdivlength
+    buf[headpos:(separation - 1)] = buf[(headpos + numsep):(headpos + numsep + headdivlength - 1)]
+
+    for i in 1:numsep
+        div = seconddiv + 3 * (i - 1)
+        buf[separation] = UInt8(',')
+        buf[(separation + 1):(separation + 3)] = buf[div:(div + 2)]
```

Comment:
we also want to avoid copying here via `getindex`

Reply:
Replaced as
```
for i in 1:numsep
        div = seconddiv + 3 * (i - 1)
        buf[separation] = UInt8(',')
        buf[(separation + 1):(separation + 3)] = view(buf, div:(div + 2))
        separation += 4
    end
```